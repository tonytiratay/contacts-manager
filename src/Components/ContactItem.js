import React, { Component } from 'react';

const isEven = num =>  num % 2 === 0;
const grey = "#efefef";

class Contactitem extends Component {

	render(){
		const { contact } = this.props;
		let style =  this.style();
		return(
			<div style={style.container}>
				<div style={style.userContainer} onClick={this.props.onClick}>
					<div style={style.imageContainer}>
						<div style={{...style.image, backgroundImage: `url(${contact.avatar})`}}></div>
					</div>
					<div style={style.userInfo}>
						<div style={style.nameContainer}>
							<h3 style={style.name}>{contact.name}</h3>
							<h3 style={style.company}>{contact.company}</h3>
						</div>
					</div>
				</div>
			</div>
		)
	}

	style(){
		return {
			container: {
				display: 'flex',
				flex: 1,
				padding: 10,
				background: isEven(this.props.index) ? '#555' : '#444',
				alignItems: 'center',
			},
			userContainer: {
				display: 'flex',
				flex: 1,
				background: isEven(this.props.index) ? '#555' : '#444',
				cursor: 'pointer',
			},
			imageContainer: {
				display: 'flex',
				marginLeft: '5px',
			},
			image: {
				width: '50px',
				height: '50px',
				alignSelf: 'center',
				backgroundSize: 'cover',
				borderRadius: '50%',
				border: isEven(this.props.index) ? '2px solid #fff' : '2px solid ' + grey,
			},
			userInfo: {
				flex: 1,
				marginLeft: '10px',
				alignItems: 'center',
				color: '#ddd',
			},
			nameContainer: {
				flex: 1,
			},
			name: {
				fontSize: '18px',
				fontWeight: '300',
				marginBottom: -10,
			},
			company: {
				fontSize: '14px',
				fontWeight: '300',
			},
		};
	}
}

export default Contactitem