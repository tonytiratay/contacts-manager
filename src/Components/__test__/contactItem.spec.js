import React from 'react';
import renderer from 'react-test-renderer';

import ContactItem from '../ContactItem';

it('Should render correctly', ()=>{

	const contact = {
		avatar: 'http://avatarurl.com/pic.jpg',
		name: 'JohnDoe',
		company: 'Leikir',
	};

	const tree = renderer
		.create(<ContactItem contact={contact} index={0} onClick={()=>{}}/>)
		.toJSON();

	expect(tree).toMatchSnapshot();
})