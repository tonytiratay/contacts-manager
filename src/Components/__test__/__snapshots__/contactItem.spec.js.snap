// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#555",
      "display": "flex",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "background": "#555",
        "cursor": "pointer",
        "display": "flex",
        "flex": 1,
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "marginLeft": "5px",
        }
      }
    >
      <div
        style={
          Object {
            "alignSelf": "center",
            "backgroundImage": "url(http://avatarurl.com/pic.jpg)",
            "backgroundSize": "cover",
            "border": "2px solid #fff",
            "borderRadius": "50%",
            "height": "50px",
            "width": "50px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "color": "#ddd",
          "flex": 1,
          "marginLeft": "10px",
        }
      }
    >
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <h3
          style={
            Object {
              "fontSize": "18px",
              "fontWeight": "300",
              "marginBottom": -10,
            }
          }
        >
          JohnDoe
        </h3>
        <h3
          style={
            Object {
              "fontSize": "14px",
              "fontWeight": "300",
            }
          }
        >
          Leikir
        </h3>
      </div>
    </div>
  </div>
</div>
`;
